### DISCUSSION TITLE
Modelling of Heat exchangers

### DISCUSSION THREAD

**Discussion by APWSvHue** [PMEAL/OpenPNM#2780] [Q&A]
Hello,

At first, thanks for this awesome python library.
I'm planning to setup a OpenPNM for a 1-D heat exchanger model.
My aim is to simulate complex fluid networks with an efficient 1D approach.
After my first investigations I think OpenPNM should be suitable for this as long as it is a laminar flow.

I want to start with a very basic with model two independent flows in a opposite direction which are separated by a solid wall.
Here is a rough sketch on what I want too set up:

![grafik](https://github.com/PMEAL/OpenPNM/assets/123973082/cb39d72a-9e77-4e5f-a0bd-cde160daabdb)

I think one way would be to have a stokes flow calculation the pressure field first and than have a second advection-diffusion run to model the thermal diffusion.
Correct?

At first I thought  about doing a discrete network like that with 3x3 pores:
![grafik](https://github.com/PMEAL/OpenPNM/assets/123973082/e224bdf6-311c-42be-9015-7805ff6a6dcc)

Pore 0,1 and 2 would be Flow 1.
Pore 3,4 and 5 would be a solid without flow rates.
Pore 6,7 and 8 would be Flow 2.

I set the hydraulic conductance of all throat except 0,3,10 and 11 to very small values to block the flow through the solid.
Furthermore, I set the 'throat.ad_dif_conductance' of 1,2,4,6,8 and 9 to a convection value (between fluid and solid) and and the 'throat.ad_dif_conductance' for 5 and 7 to a conduction value (inside the solid material).

![grafik](https://github.com/PMEAL/OpenPNM/assets/123973082/85c18113-874a-49e1-a98e-2407ed694a1a)

My first results doesn't look to bad (see the picture above Flow 1 is cooled from 400K to 325 K, Flow 2 is heated from 300K to 375 K) .
But do you think that is a good overall approach?
Or is there another good way to model it with OpenPNM?

Thanks and best regards

**Comment by jgostick** [comment]
This seems like a good approach, except I would say you should use a lot more 'nodes' along the channel to get a better approximation.  The computational cost will still be negligible.  This will let you have more places for heat to cross over.  Although, what you're doing is probably just as easily done in comosl no?  Do you have a more complicated domain planned for the future?

**Reply by jgostick** [reply]
I should also add that we use OpenPNM almost exclusively for mass transfer problems.  So although heat transfer should be do-able, there is a chance that something is quite right with heat-transfer advection-diffusion, or some other bug.

**Reply by APWSvHue** [reply]
Hi,

Thanks for your fast feedback!
If you think it's a good approach I will progress like that.

Yes I plan to use it for more complicated domains in the future.
The Idea is to replace complicated CFD models with very simple 1D models.
If I have troubles with the correlation between CFD and OpenPNM I will come back to the forum.

BR.

**Comment by jundonglin** [comment]
Hello,
I am very interested in your simulation, I did not understand something in the process of similar simulation,.
I would appreciate it if you could send me your code to study.
